packages/core/admin/ee/admin/pages/SettingsPage/pages/AuditLogs/ListView/hooks/useAuditLogsData.js

```diff
@@ -2,49 +2,57 @@ import { useQuery } from 'react-query';
 import { useNotification, useFetchClient } from '@strapi/helper-plugin';
 import { useLocation } from 'react-router-dom';
 
+import { useAdminUsers } from '../../../../../../../../admin/src/hooks/useAdminUsers';
+
 const useAuditLogsData = ({ canReadAuditLogs, canReadUsers }) => {
   const { get } = useFetchClient();
   const { search } = useLocation();
   const toggleNotification = useNotification();
 
-  const fetchAuditLogsPage = async ({ queryKey }) => {
-    const search = queryKey[1];
-    const { data } = await get(`/admin/audit-logs${search}`);
-
-    return data;
-  };
-
-  const fetchAllUsers = async () => {
-    const { data } = await get(`/admin/users`);
-
-    return data;
-  };
-
   const queryOptions = {
     keepPreviousData: true,
     retry: false,
     staleTime: 1000 * 20, // 20 seconds
     onError: (error) => toggleNotification({ type: 'warning', message: error.message }),
   };
 
+  const {
+    users,
+    isError: isUsersError,
+    isLoading: isLoadingUsers,
+  } = useAdminUsers(
+    {},
+    {
+      ...queryOptions,
+      enabled: canReadUsers,
+      staleTime: 2 * (1000 * 60), // 2 minutes
+    }
+  );
+
   const {
     data: auditLogs,
-    isLoading,
+    isLoading: isLoadingAuditLogs,
     isError: isAuditLogsError,
-  } = useQuery(['auditLogs', search], fetchAuditLogsPage, {
-    ...queryOptions,
-    enabled: canReadAuditLogs,
-  });
-
-  const { data: users, isError: isUsersError } = useQuery(['auditLogsUsers'], fetchAllUsers, {
-    ...queryOptions,
-    enabled: canReadUsers,
-    staleTime: 2 * (1000 * 60), // 2 minutes
-  });
-
-  const hasError = isAuditLogsError || isUsersError;
-
-  return { auditLogs, users: users?.data, isLoading, hasError };
+  } = useQuery(
+    ['auditLogs', search],
+    async ({ queryKey }) => {
+      const search = queryKey[1];
+      const { data } = await get(`/admin/audit-logs${search}`);
+
+      return data;
+    },
+    {
+      ...queryOptions,
+      enabled: canReadAuditLogs,
+    }
+  );
+
+  return {
+    auditLogs,
+    users,
+    isLoading: isLoadingUsers || isLoadingAuditLogs,
```

Comment:
The users for audit logs are only used in the filters so I think we left them out of isLoading to not block the UI and just load them in the background. wdyt? Do you still prefer to consider them in the loading state? @remidej correct me if I'm wrong but I think that was our idea.

Reply:
I don't mind reverting it, but it seems more save to me, because you are assuming on request finishes before the other, which might not be always true.

Reply:
works for me, better safe than sorry :)